### PR TITLE
[mle] add helper 'InitNetifUnicastAddress()'

### PR DESCRIPTION
**Pull request by abtink**

### CODE REVIEW

File: src/core/thread/mle.cpp [hunk 1]
```diff
@@ -200,6 +173,17 @@ Mle::Mle(Instance &aInstance)
     // to the Link- and Realm-Local All Thread Nodes multicast addresses.
 }
 
+void Mle::InitNetifUnicastAddress(Ip6::NetifUnicastAddress &aAddress)
```
Comment (jwhui) [inline src/core/thread/mle.cpp:176]: Should we also leverage this for other components? For example:
https://github.com/openthread/openthread/blob/7dc3b2696ade4a524b8e2e33cd73bed2668af408/src/core/meshcop/border_agent.cpp#L355-L361

File: src/core/thread/mle.hpp [hunk 2]
```diff
@@ -1622,6 +1622,7 @@ class Mle : public InstanceLocator
         TimeMilli    mSendTime;    // Time when the message shall be sent.
     };
 
+    void        InitNetifUnicastAddress(Ip6::NetifUnicastAddress &aAddress);
```
Comment (simonlingoogle) [inline src/core/thread/mle.hpp:1625]: make this method static?

File: src/core/thread/mle.cpp (hunk 1 above)
Comment (abtink) [reply src/core/thread/mle.cpp:176]: Thinking more on this, I think may be best to add one or more such `Init()` methods in `NetifUnicastAddress` class itself. This way can be used by all other classes.

**Comment by abtink** [comment]
Closing this PR in favor of PR https://github.com/openthread/openthread/pull/5390.

**Pull request closed without merging**